### PR TITLE
Shodan v2

### PR DESCRIPTION
**Pull request by Shakedp**
## Status
Ready 

## Related Issues
https://github.com/demisto/etc/issues/17477

## Description
Shodan integration enhancement

## Required version of Demisto
5.0.0

## Does it break backward compatibility?
   - No

## Must have
- [ x ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/Shodan/Shodan.py [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+    demisto.results(f'Deleted alert {alert_id}')
+
+
+def shodan_network_alert_set_trigger_command():
+    alert_id = demisto.args()['alertID']
+    trigger = demisto.args()['Trigger']
+
+    res = http_request('PUT', f'/shodan/alert/{alert_id}/trigger/{trigger}')
+
+    if 'success' not in res or res['success'] is not True:
```
Comment (hirshbergshachar) [inline Integrations/Shodan/Shodan.py]: res.get('success', False)

File: Integrations/Shodan/Shodan.py [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    demisto.results(f'Set trigger "{trigger}" for alert {alert_id}')
+
+
+def shodan_network_alert_remove_trigger_command():
+    alert_id = demisto.args()['alertID']
+    trigger = demisto.args()['Trigger']
+
+    res = http_request('DELETE', f'/shodan/alert/{alert_id}/trigger/{trigger}')
+
+    if 'success' not in res or res['success'] is not True:
```
Comment (hirshbergshachar) [inline Integrations/Shodan/Shodan.py]: res.get('success', False)

File: Integrations/Shodan/Shodan.py [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+
+
+def shodan_network_alert_whitelist_service_command():
+    alert_id = demisto.args()['alertID']
+    trigger = demisto.args()['trigger']
+    service = demisto.args()['service']
+
+    res = http_request('PUT', f'/shodan/alert/{alert_id}/trigger/{trigger}/ignore/{service}')
+
+    if 'success' not in res or res['success'] is not True:
```
Comment (hirshbergshachar) [inline Integrations/Shodan/Shodan.py]: res.get('success', False)

File: Integrations/Shodan/Shodan.py [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+
+
+def shodan_network_alert_remove_service_from_whitelist_command():
+    alert_id = demisto.args()['alertID']
+    trigger = demisto.args()['trigger']
+    service = demisto.args()['service']
+
+    res = http_request('DELETE', f'/shodan/alert/{alert_id}/trigger/{trigger}/ignore/{service}')
+
+    if 'success' not in res or res['success'] is not True:
```
Comment (hirshbergshachar) [inline Integrations/Shodan/Shodan.py]: res.get('success', False)

File: Integrations/Shodan/Shodan.py (hunk 1 above)
Comment (Shakedp) [reply Integrations/Shodan/Shodan.py]: Huh, yeah that makes much more sense:)

**Comment by kirbles19** [comment]
Link to what I'm supposed to review?

**Comment by Itay4** [comment]
@hirshbergshachar how do we want to progress here?

**Comment by hirshbergshachar** [comment]
@kirbles19 did you finish your adds? Code is complete...

**Comment by kirbles19** [comment]
Sure, didn't know it was ready to go. We'll have it ready for next content release.

**Comment by hirshbergshachar** [comment]
@kirbles19 great. Please ping me here after you finish

**Comment by kirbles19** [comment]
@hirshbergshachar If Yaniv has questions about the integration, who should he meet with (Shaked is still away).

**Comment by hirshbergshachar** [comment]
@kirbles19 @ygerowitz I'll answer the questions

**Comment by ygerowitz** [comment]
@hirshbergshachar I have a few questions that I will send to you. Thanks.

**Comment by ygerowitz** [comment]
Reviewed document.

**Comment by hirshbergshachar** [comment]
@ygerowitz don't close the pull request please, the branch wasn't merged...

**Pull request merged**